guides/release-checklist.md

```diff
@@ -0,0 +1,116 @@
+# Inputs (Preparation Tasks):
+
+- Candidate commit for the release, with the desired state of the app, on `master` branch (make sure the app is actually ready for release)
+- Changelog up to date with changes since last release
+- Blurb text (some flavor text and info about what happened this release, for GitHub release and blog post)
+
+# Outputs:
+
+- Version bump PR
+- Changelog (within the welcome package + GitHub Release and blog post)
+- GitHub Release
+  - With binaries
+- Update download links on website
+- Blog Post
+- Social posts
+  - Discord announcement
+  - Reddit post
+  - Mastodon post
+
+# Steps:
+
+Note: after the "Changelog" steps are done, the remaining steps can be done in whatever order.
+
+- Prepare the Changelog.
+  - We should make sure last-minute PRs being included in the release are also in the changelog.
+
+- Update the Changelog in-app (in the `welcome` package)
+  - Example PR: https://github.com/pulsar-edit/pulsar/pull/489
+
+
+- Do the version bump pull request.
+  - (Open this once fairly confident the app is ready to roll out, after the in-app Changelog PR is merged -- make sure to include any last-minute changes. If more last-minute changes happen, update the PR and potentially the tag, see next bullet point.)
+  - This should PR be opened from a branch up to date with `master`, plus one commit which changes the version in package.json to the version you're going to release.
+    - Tag that "version update" commit with the tag name starting with v and then the version number of the release. (For example v1.103.0)
+  - Let CI run, grab the generated binaries from Cirrus, since we're going to upload these to GitHub Releases as well.
+  - Before actually merging this PR, change the version number to end with -dev again, for the sake of the Rolling releases that will build later on `master` branch.
+  - Example PR: https://github.com/pulsar-edit/pulsar/pull/430
+
+- Do the GitHub release, suggested to include the blurb text and the changelog.
+  - Can start drafting this any time.
+  - Recommended to only have one person editing it at once, to avoid overwrites and lost edits
+  - Once we have binaries generated (see the "pull request" steps above), someone can start downloading these and re-uploading them to GitHub Releases under the draft release
+  - Post the GitHub release! 🎉 
+  - Example release: https://github.com/pulsar-edit/pulsar/releases
+
+- Draft and post the blog post, suggested to be based on the blurb text plus the changelog.
+  - See past blog post PRs for examples
+  - Example PR: https://github.com/pulsar-edit/pulsar-edit.github.io/pull/204
+
+- Update the download links on the website
+  - This will at least need updates for the new version number.
+  - Recommended to verify the exact filenames of all binaries, in case some of them may have changed, such as if new or different formats were built this release. Adjust the download page as-needed if new/different formats were built than last time.
```

Comment:
```suggestion
  - Recommended to verify the exact filenames of all binaries, in case some of them may have changed, such as if new or different formats were built this release. Adjust the download page as-needed if new/different formats were built than last time.
  - Update the `Current version is...` line to the new version
```

Honestly I forget this step all the time so worth adding here I think. I'm referring to this line: https://github.com/pulsar-edit/pulsar-edit.github.io/blob/7262ce81685925611a94acf5cf31e29756ea01fa/docs/download.md?plain=1#L115

Reply:
👌